nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -409,7 +404,20 @@ bool RegulatedPurePursuitController::isCollisionImminent(
   pose_msg.header.frame_id = arc_pts_msg.header.frame_id;
   pose_msg.header.stamp = arc_pts_msg.header.stamp;
 
-  const double projection_time = costmap_->getResolution() / fabs(linear_vel);
+  double projection_time = 0.0;
+  if (fabs(linear_vel) < 0.01 && fabs(angular_vel) > 0.01) {
+    // rotating to heading at goal or toward path
+    // Equation finds the angular distance required for the largest
+    // part of the robot radius to move to another costmap cell:
+    // theta_min = 2.0 * sin ((res/2) / r_max)
+    // via isosceles triangle r_max-r_max-resolution,
+    // dividing by angular_velocity gives us a timestep.
+    projection_time =
+      2.0 * sin((costmap_->getResolution() / 2) / 0.25 /*TODO radius*/) / fabs(angular_vel);
+  } else {
+    // Normal path tracking
+    projection_time = costmap_->getResolution() / fabs(linear_vel);
```

Comment:
Shouldn't we check that ```linear_vel``` is not null to prevent division-by-zero ?

Reply:
There is no time that is possible if you look at the code. There are tons of minimum parameterizations set such that it cannot be 0.0 ever, plus if it were 0.0, `0.0 * curvature` would be nonsensical so the entire algorithm would fail to operate

Reply:
I suppose it could happen if people misset the parameters, I'll make a note of that in the configuration guide.